cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/GrantFetcher.java

```diff
@@ -19,18 +19,20 @@
 import io.cdap.cdap.api.security.AccessException;
 import io.cdap.cdap.proto.security.GrantedPermission;
 import io.cdap.cdap.proto.security.Principal;
+
+import java.util.Map;
 import java.util.Set;
 
 /**
  * Fetches {@link GrantedPermission grants} of the specified {@link Principal}.
  */
-public interface GrantFetcher {
+public interface GrantFetcher<T> {
```

Comment:
We can't change any existing interfaces in backward-incompatible way. Only changes allowed are new methods with default implementation